### DISCUSSION TITLE
Does tailwind css supports define the font size xs values at each screen sizes level

### DISCUSSION THREAD

**Discussion by rahultokase** [tailwindlabs/tailwindcss#11763] [General]
currently, we can customize the font size values by extending it and it gets applied to all the applicable screen sizes by default. Are we thinking of supporting the extension the font size value per screen size basis ?

**Comment by torressam333** [comment]
I wish tailwind has an `xs:text-4xl` option. Any screens under 640px needs custom css media queries.

**Reply by andymccray** [reply]
You can use an un-prefixed rule to target screens under 640px:

```html
<h1 class="text-4xl sm:text-5xl">…</div>
```

Which says to apply the `text-4xl` from 0px  to 639px and then use `text-5xl` once the screen hits 640px. You can also define your own custom screen sizes if you need more granular control between 0 and 640px.

**Reply by wongjn** [reply]
To add to this, if only "simple" `screens` are defined in the Tailwind configuration, one can also use [`max-*:` variants](https://tailwindcss.com/docs/responsive-design#targeting-a-breakpoint-range) like:

```html
<h1 class="max-sm:text-4xl text-5xl">…</div>
```
This would be equivalent to the previous snippet.

Furthermore, [arbitrary media query variants exist too](https://tailwindcss.com/docs/responsive-design#arbitrary-values).

**Comment by torressam333** [comment]
Thanks, @andymccray !
